posthog/temporal/data_modeling/run_workflow.py

```diff
@@ -337,7 +360,45 @@ async def materialize_model(model_label: str, team: Team) -> tuple[str, DeltaTab
         prepare_s3_files_for_querying(saved_query.folder_path, saved_query.name, file_uris)
 
     key, delta_table = tables.popitem()
-    return (key, delta_table)
+
+    # Count rows and update both DataWarehouseTable and DataModelingJob
+    row_count = await asyncio.to_thread(count_delta_table_rows, delta_table)
+    await update_table_row_count(saved_query, row_count)
+
+    # Update the job record with the row count and completed status
+    job.rows_materialized = row_count
+    job.status = DataModelingJob.Status.COMPLETED
+    job.last_run_at = dt.datetime.now(dt.UTC)
+    await database_sync_to_async(job.save)()
+
+    return (key, delta_table, job.id)
+
+
+def count_delta_table_rows(delta_table: DeltaTable) -> int:
+    """
+    Count the number of rows in a Delta table using metadata.
+    """
+    count = 0
+    for batch in delta_table.to_pyarrow_dataset().to_batches():
+        count += len(batch)
+    return count
+
+
+async def update_table_row_count(saved_query: DataWarehouseSavedQuery, row_count: int) -> None:
+    """Update the row count in the DataWarehouseTable record. `saved_query` name is unique per team."""
+    try:
+        table = await database_sync_to_async(
+            DataWarehouseTable.objects.filter(team_id=saved_query.team_id, name=saved_query.name).first
```

Comment:
`DataWarehouseSavedQuery` has a `table_id` property that can be used instead here:

```python
DataWarehouseTable.objects.get(team_id=saved_query.team_id, id=saved_query.table_id)
```